Tests/Contour.Common.Tests/AbstactSenderSpecs.cs

```diff
@@ -0,0 +1,135 @@
+                            TaskCompletionSource<MessageExchange> tcs = new TaskCompletionSource<MessageExchange>();
+                            tcs.SetResult(me);
+                            return tcs.Task;
+                        },
+                    new Mock<IEndpoint>().Object,
+                    senderConfigMock.Object,
+                    Enumerable.Empty<IMessageExchangeFilter>());
+
+                var r = sut.Request<object>(MessageLabel.Empty, new { }, new RequestOptions()).Result;
+
+                Assert.IsNotNull(sentMessage);
+                Assert.AreEqual(incomingHeaders[Headers.OriginalMessageId], sentMessage.Headers[Headers.OriginalMessageId]);
+                Assert.AreNotEqual(incomingHeaders[Headers.CorrelationId], sentMessage.Headers[Headers.CorrelationId]);
+            }
+
+            [Test]
+            public void ShouldReplaceCorrelationIdOnRequestWithoutLabel()
+            {
+                var storageMock = new Mock<IIncomingMessageHeaderStorage>();
+                var incomingHeaders = new Dictionary<string, object>
+                {
+                    [Headers.OriginalMessageId] = "OriginalMessageId",
+                    [Headers.CorrelationId] = "CorrelationId"
+                };
+                storageMock.Setup(s => s.Load())
+                    .Returns(
+                        incomingHeaders);
+                var senderConfigMock = new Mock<ISenderConfiguration>();
+                senderConfigMock.Setup(s => s.Options)
+                    .Returns(
+                        new SenderOptions
+                        {
+                            IncomingMessageHeaderStorage = new Maybe<IIncomingMessageHeaderStorage>(storageMock.Object)
+                        });
+                senderConfigMock.Setup(s => s.Label)
+                    .Returns(MessageLabel.Any);
+
+                IMessage sentMessage = null;
+                var sut = new TestSender(
+                    me =>
+                    {
+                        sentMessage = me.Out;
+                        me.In = new Message(MessageLabel.Any, new object());
+                        TaskCompletionSource<MessageExchange> tcs = new TaskCompletionSource<MessageExchange>();
+                        tcs.SetResult(me);
+                        return tcs.Task;
+                    },
+                    new Mock<IEndpoint>().Object,
+                    senderConfigMock.Object,
+                    Enumerable.Empty<IMessageExchangeFilter>());
+
+                var r = sut.Request<object>(new { }, new RequestOptions()).Result;
+
+                Assert.IsNotNull(sentMessage);
+                Assert.AreEqual(incomingHeaders[Headers.OriginalMessageId], sentMessage.Headers[Headers.OriginalMessageId]);
+                Assert.AreNotEqual(incomingHeaders[Headers.CorrelationId], sentMessage.Headers[Headers.CorrelationId]);
+            }
+        }
+
+        private class TestSender : AbstractSender
```

Comment:
In this particular case the stub type does not provide the implementation to be tested it is already provided by Abstract Sender.